src/java.base/share/classes/java/lang/Float16.java

```diff
@@ -0,0 +1,1409 @@
+     *         argument.
+     * @throws NullPointerException  if the string is null
+     * @throws NumberFormatException if the string does not contain a
+     *               parsable {@code Float16}.
+     * @see    java.lang.Float#valueOf(String)
+     */
+    public static Float16 valueOf(String s) throws NumberFormatException {
+        // TOOD: adjust precision of parsing if needed
+        return valueOf(Double.parseDouble(s));
+    }
+
+    /**
+     * {@return a {@link Float16} value rounded from the {@link BigDecimal}
+     * argument using the round to nearest rounding policy}
+     *
+     * @apiNote
+     * This method corresponds to the convertFormat operation defined
+     * in IEEE 754.
+     *
+     * @param  v a {@link BigDecimal}
+     * @see java.math.BigDecimal#float16Value()
+     */
+    public static Float16 valueOf(BigDecimal v) {
+        return v.float16Value();
+    }
+
+    /**
+     * Returns {@code true} if the specified number is a
+     * Not-a-Number (NaN) value, {@code false} otherwise.
+     *
+     * @apiNote
+     * This method corresponds to the isNaN operation defined in IEEE
+     * 754.
+     *
+     * @param   f16   the value to be tested.
+     * @return  {@code true} if the argument is NaN;
+     *          {@code false} otherwise.
+     *
+     * @see Float#isNaN(float)
+     * @see Double#isNaN(double)
+     */
+    public static boolean isNaN(Float16 f16) {
+        final short bits = float16ToRawShortBits(f16);
+        // A NaN value has all ones in its exponent and a non-zero significand
+        return ((bits & 0x7c00) == 0x7c00 && (bits & 0x03ff) != 0);
+    }
+
+    /**
+     * Returns {@code true} if the specified number is infinitely
+     * large in magnitude, {@code false} otherwise.
+     *
+     * @apiNote
+     * This method corresponds to the isInfinite operation defined in
+     * IEEE 754.
+     *
+     * @param   f16   the value to be tested.
+     * @return  {@code true} if the argument is positive infinity or
+     *          negative infinity; {@code false} otherwise.
+     *
+     * @see Float#isInfinite(float)
+     * @see Double#isInfinite(double)
+     */
+    public static boolean isInfinite(Float16 f16) {
+        return ((float16ToRawShortBits(f16) ^ float16ToRawShortBits(POSITIVE_INFINITY)) & 0x7fff) == 0;
+    }
+
+    /**
+     * Returns {@code true} if the argument is a finite floating-point
+     * value; returns {@code false} otherwise (for NaN and infinity
+     * arguments).
+     *
+     * @apiNote
+     * This method corresponds to the isFinite operation defined in
+     * IEEE 754.
+     *
+     * @param f16 the {@code Float16} value to be tested
+     * @return {@code true} if the argument is a finite
+     * floating-point value, {@code false} otherwise.
+     *
+     * @see Float#isFinite(float)
+     * @see Double#isFinite(double)
+     */
+    public static boolean isFinite(Float16 f16) {
```

Comment:
Hi Jatin, are we not including intrinsic support for isFinite/Infinite/NaN as well in this PR?